src/extension.ts

```diff
@@ -1031,16 +1051,38 @@ function syncKubernetes() {
     });
 }
 
-const deleteKubernetes = () => {
-    findKindNameOrPrompt(kuberesources.commonKinds, 'delete', { nameOptional: true }, (kindName) => {
-        if (kindName) {
-            let commandArgs = kindName;
-            if (!containsName(kindName)) {
-                commandArgs = kindName + " --all";
-            }
-            kubectl.invoke('delete ' + commandArgs);
+const deleteKubernetes = async (explorerNode? : explorer.ResourceNode) => {
+    if (explorerNode) {
+        const answer = await vscode.window.showWarningMessage(`Do you want to delete the resource '${explorerNode.resourceId}'?`, ...deleteMessageItems);
+        if (answer.isCloseAffordance) {
+            return;
         }
-    });
+        const shellResult = await kubectl.invokeAsyncWithProgress(`delete ${explorerNode.resourceId}`, `Deleting ${explorerNode.resourceId}...`);
+        if (shellResult.code !== 0) {
```

Comment:
Lines 1061-1067 look identical to lines 1076ff.  Should we extract this into a `reportDeleteResult` function perhaps?